ui/pages/send/send-content/send-content.component.js

```diff
@@ -44,13 +46,18 @@ export default class SendContent extends Component {
       isAssetSendable,
       networkOrAccountNotSupports1559,
       getIsBalanceInsufficient,
+      asset,
     } = this.props;
 
     let gasError;
     if (gasIsExcessive) gasError = GAS_PRICE_EXCESSIVE_ERROR_KEY;
     else if (noGasPrice) gasError = GAS_PRICE_FETCH_FAILURE_ERROR_KEY;
     else if (getIsBalanceInsufficient)
       gasError = INSUFFICIENT_FUNDS_FOR_GAS_ERROR_KEY;
+    const showHexData =
+      this.props.showHexData &&
+      asset.type !== ASSET_TYPES.NATIVE &&
```

Comment:
Why don't we want to showHexData is the asset type is `NATIVE`?

Reply:
When I added in hex data for an Ether send it no longer showed the transaction as a transfer on etherscan. I'm also not sure why we would want to allow hex data for a `NATIVE` token send, open to hearing your thoughts :)

![image](https://user-images.githubusercontent.com/41640681/140065326-96b81ec1-720b-4b9d-ad75-418909999e6c.png)

Reply:
But... this would disable custom hex data for all transactions :sweat_smile: There are only two types - native and token.

The _sole purpose_ of this field is to add additional data to simple send transactions. Sometimes people want to attach messages to their transfers, and sometimes they just want a way to embed a message on the blockchain. Maybe there are other reasons, I'm not sure.

Either way, I don't see why we'd care how Etherscan displays such transactions.

Reply:
Okay, yeah, you can just hide hex data if `asset.type !== ASSET_TYPES.TOKEN`

Reply:
@Gudahtt That's true, I suppose hex data wouldn't matter for a native token send, I hadn't realized we were not wanting to enable hex data for ERC-20 was because the hex data would potentially mess up the transfer invocation 😅. A transfer can be successful with hex data if the ERC20 has extra functions with an extra field on `transfer` but I guess that would be too much of a pain to deal with. Update coming in a few!

Reply:
> To clarify that a little bit, it's not so much that it would mess up the transfer invocation. But rather that right now we _overwrite_ the custom hex data entered by the user with the transfer invocation. So it's misleading to include the filed when it just gets overwritten.

Good to know, thanks!